### PR TITLE
Add automation icon

### PR DESCRIPTION
**Pull request by martenbjork**
Designed by @AdamWhitcroft, PR by me.

### CODE REVIEW

File: packages/polaris-icons-raw/icons/polaris/automation_major.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: automation
+set: major
+authors:
+  - Adam Whitcroft
+description: Used to represent automation, or a task that can be automated.
+keywords:
+  - automation
+  - cog
+  - automated
```
Comment (amrocha) [inline packages/polaris-icons-raw/icons/polaris/automation_major.yml]: gear?

**Comment by amrocha** [comment]
CI is gonna fail and complain about a bunch of things, fix those things and you should be good 👍

File: packages/polaris-icons-raw/icons/polaris/automation_major.yml (hunk 1 above)
Comment (martenbjork) [reply packages/polaris-icons-raw/icons/polaris/automation_major.yml]: Good one, adding.

Comment (AdamWhitcroft) [reply packages/polaris-icons-raw/icons/polaris/automation_major.yml]: We don't use `Gear` for https://polaris-icons.shopify.com/?icon=SettingsMajor&q=settings - should that be updated too?

File: packages/polaris-icons-raw/icons/polaris/automation_major_twotone.svg [hunk 2]
```diff
@@ -0,0 +1 @@
...6789 8.99995367,16.7100873 L8.99997977,18 L11.0002952,18 L11.0006762,16.7096248 C11.0008068,16.2673532 11.2914454,15.8776966 11.7153292,15.7514958 C12.6164265,15.4832164 13.4400914,15.0050211 14.122343,14.3596436 C14.4436729,14.0556809 14.92652,13.9989285 15.3095726,14.2201001 L16.428219,14.8659981 L17.4282045,13.1339724 L16.3109127,12.4889 C15.928172,12.2679233 15.7358769,11.8219096 15.8379535,11.3919079 C15.9452458,10.9399349 16,10.4739329 16,10 C16,9.52606708 15.9452458,9.06006506 15.8379535,8.60809214 C15.7358769,8.17809042 15.928172,7.73207674 16.3109127,7.51110004 L17.4282045,6.86602758 L16.428219,5.13400195 L15.3095726,5.77989986 C14.92652,6.00107153 14.4436729,5.94431911 14.122343,5.64035639 C13.4400914,4.99497892 12.6164265,4.51678356 11.7153292,4.24850424 C11.2914454,4.12230337 11.0008068,3.73264679 11.0006762,3.29037524 L11.0002952,2 Z M8,8 C8,7.20130462 8.89014547,6.72491322 9.5547002,7.16794971 L9.5547002,7.16794971 L12.5547002,9.16794971 C13.1484333,9.56377175 13.1484333,10.4362282 12.5547002,10.8320503 L12.5547002,10.8320503 L9.5547002,12.8320503 C8.89014547,13.2750868 8,12.7986954 8,12 L8,12 Z" id="Combined-Shape" fill="#637381" fill-rule="nonzero"></path></g></svg>
```
Comment (kaelig) [inline packages/polaris-icons-raw/icons/polaris/automation_major_twotone.svg]:
```suggestion
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20"><path fill="#637381" d="M12 0a1 1 0 011 1v1.58a8 8 0 011.92 1.11l1.37-.79a1 1 0 011.37.37l2 3.46a1 1 0 01-.37 1.37l-1.37.79a8.06 8.06 0 010 2.22l1.37.8a1 1 0 01.37 1.36l-2 3.46a1 1 0 01-1.37.37l-1.37-.8A8 8 0 0113 17.43V19a1 1 0 01-1 1H8a1 1 0 01-1-1v-1.58a8 8 0 01-1.92-1.11l-1.37.79a1 1 0 01-1.37-.37l-2-3.46a1 1 0 01.37-1.37l1.37-.79a8.06 8.06 0 010-2.22L.7 8.09a1 1 0 01-.37-1.36l2-3.46A1 1 0 013.7 2.9l1.37.8a8 8 0 011.58-.97L7 2.58V1a1 1 0 011-1h4zm-1 2H9v1.29a1 1 0 01-.71.96 6 6 0 00-2.41 1.39 1 1 0 01-1.19.14l-1.12-.65-1 1.74 1.12.64a1 1 0 01.47 1.1 6.02 6.02 0 000 2.78 1 1 0 01-.47 1.1l-1.12.64 1 1.74 1.12-.65a1 1 0 011.19.14 6 6 0 002.4 1.4 1 1 0 01.72.95V18h2v-1.29a1 1 0 01.72-.96 6 6 0 002.4-1.39 1 1 0 011.19-.14l1.12.65 1-1.74-1.12-.64a1 1 0 01-.47-1.1 6.02 6.02 0 000-2.78 1 1 0 01.47-1.1l1.12-.64-1-1.74-1.12.65a1 1 0 01-1.19-.14 6 6 0 00-2.4-1.4A1 1 0 0111 3.3V2zM8 8a1 1 0 011.55-.83l3 2a1 1 0 010 1.66l-3 2A1 1 0 018 12z"/></svg>

```

File: packages/polaris-icons-raw/icons/polaris/automation_major_monotone.svg [hunk 3]
```diff
@@ -0,0 +1 @@
...6789 8.99995367,16.7100873 L8.99997977,18 L11.0002952,18 L11.0006762,16.7096248 C11.0008068,16.2673532 11.2914454,15.8776966 11.7153292,15.7514958 C12.6164265,15.4832164 13.4400914,15.0050211 14.122343,14.3596436 C14.4436729,14.0556809 14.92652,13.9989285 15.3095726,14.2201001 L16.428219,14.8659981 L17.4282045,13.1339724 L16.3109127,12.4889 C15.928172,12.2679233 15.7358769,11.8219096 15.8379535,11.3919079 C15.9452458,10.9399349 16,10.4739329 16,10 C16,9.52606708 15.9452458,9.06006506 15.8379535,8.60809214 C15.7358769,8.17809042 15.928172,7.73207674 16.3109127,7.51110004 L17.4282045,6.86602758 L16.428219,5.13400195 L15.3095726,5.77989986 C14.92652,6.00107153 14.4436729,5.94431911 14.122343,5.64035639 C13.4400914,4.99497892 12.6164265,4.51678356 11.7153292,4.24850424 C11.2914454,4.12230337 11.0008068,3.73264679 11.0006762,3.29037524 L11.0002952,2 Z M8,8 C8,7.20130462 8.89014547,6.72491322 9.5547002,7.16794971 L9.5547002,7.16794971 L12.5547002,9.16794971 C13.1484333,9.56377175 13.1484333,10.4362282 12.5547002,10.8320503 L12.5547002,10.8320503 L9.5547002,12.8320503 C8.89014547,13.2750868 8,12.7986954 8,12 L8,12 Z" id="Combined-Shape" fill="#637381" fill-rule="nonzero"></path></g></svg>
```
Comment (kaelig) [inline packages/polaris-icons-raw/icons/polaris/automation_major_monotone.svg]:
```suggestion
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20"><path fill="#637381" d="M12 0a1 1 0 011 1v1.58a8 8 0 011.92 1.11l1.37-.79a1 1 0 011.37.37l2 3.46a1 1 0 01-.37 1.37l-1.37.79a8.06 8.06 0 010 2.22l1.37.8a1 1 0 01.37 1.36l-2 3.46a1 1 0 01-1.37.37l-1.37-.8A8 8 0 0113 17.43V19a1 1 0 01-1 1H8a1 1 0 01-1-1v-1.58a8 8 0 01-1.92-1.11l-1.37.79a1 1 0 01-1.37-.37l-2-3.46a1 1 0 01.37-1.37l1.37-.79a8.06 8.06 0 010-2.22L.7 8.09a1 1 0 01-.37-1.36l2-3.46A1 1 0 013.7 2.9l1.37.8a8 8 0 011.58-.97L7 2.58V1a1 1 0 011-1h4zm-1 2H9v1.29a1 1 0 01-.71.96 6 6 0 00-2.41 1.39 1 1 0 01-1.19.14l-1.12-.65-1 1.74 1.12.64a1 1 0 01.47 1.1 6.02 6.02 0 000 2.78 1 1 0 01-.47 1.1l-1.12.64 1 1.74 1.12-.65a1 1 0 011.19.14 6 6 0 002.4 1.4 1 1 0 01.72.95V18h2v-1.29a1 1 0 01.72-.96 6 6 0 002.4-1.39 1 1 0 011.19-.14l1.12.65 1-1.74-1.12-.64a1 1 0 01-.47-1.1 6.02 6.02 0 000-2.78 1 1 0 01.47-1.1l1.12-.64-1-1.74-1.12.65a1 1 0 01-1.19-.14 6 6 0 00-2.4-1.4A1 1 0 0111 3.3V2zM8 8a1 1 0 011.55-.83l3 2a1 1 0 010 1.66l-3 2A1 1 0 018 12z"/></svg>

```

**Comment by kaelig** [comment]
I went ahead and optimized the icons in https://jakearchibald.github.io/svgomg/!

**Comment by martenbjork** [comment]
> I went ahead and optimized the icons in https://jakearchibald.github.io/svgomg/!

Thanks @kaelig! I missed your fix and learned the linting rules the hard way by running `npm run test` over and over 😊 Good learning opportunity though. Let's see this passes build tests.

File: packages/polaris-icons-raw/icons/polaris/automation_major.yml (hunk 1 above)
Comment (amrocha) [reply packages/polaris-icons-raw/icons/polaris/automation_major.yml]: Yeah, these keywords just help with search so I think that makes sense

**Comment by kaelig** [comment]
We're missing a few things in our test suite:

- There shouldn't be an `<?xml version="1.0" encoding="UTF-8"?>`
- missing line at end of file
- 7-digit precision is too high, we should cap it to 3

**Comment by amrocha** [comment]
>There shouldn't be an <?xml version="1.0" encoding="UTF-8"?>
missing line at end of file
7-digit precision is too high, we should cap it to 3

The first two get optimized away before build, and if you want you can configure SVGO to fix the third one for you. And you can automate the tests to catch the first two as well.

It'd be great if you could do that, you'd save everyone a lot of time and help make the contribution process smoother.

**Comment by martenbjork** [comment]
And you can automate the tests to catch the first two as well.

@amrocha @kaelig I took a stab at this: https://github.com/Shopify/polaris-icons/commit/f58d116deeec5a6495384b08fb2a672f75e0fc46

This would have caught my errors. What do you think?

**Comment by amrocha** [comment]
LGTM, make sure to check that the tests actually fail if you try to run them on your original SVG

**Comment by martenbjork** [comment]
✅ They do!

File: packages/polaris-icons-raw/icons/polaris/automation_major_monotone.svg [hunk 4]
```diff
@@ -0,0 +1 @@
...3,15.877856 8.99994472,16.2676789 8.99995367,16.7100873 L8.99997977,18 L11.0002952,18 L11.0006762,16.7096248 C11.0008068,16.2673532 11.2914454,15.8776966 11.7153292,15.7514958 C12.6164265,15.4832164 13.4400914,15.0050211 14.122343,14.3596436 C14.4436729,14.0556809 14.92652,13.9989285 15.3095726,14.2201001 L16.428219,14.8659981 L17.4282045,13.1339724 L16.3109127,12.4889 C15.928172,12.2679233 15.7358769,11.8219096 15.8379535,11.3919079 C15.9452458,10.9399349 16,10.4739329 16,10 C16,9.52606708 15.9452458,9.06006506 15.8379535,8.60809214 C15.7358769,8.17809042 15.928172,7.73207674 16.3109127,7.51110004 L17.4282045,6.86602758 L16.428219,5.13400195 L15.3095726,5.77989986 C14.92652,6.00107153 14.4436729,5.94431911 14.122343,5.64035639 C13.4400914,4.99497892 12.6164265,4.51678356 11.7153292,4.24850424 C11.2914454,4.12230337 11.0008068,3.73264679 11.0006762,3.29037524 L11.0002952,2 Z M8,8 C8,7.20130462 8.89014547,6.72491322 9.5547002,7.16794971 L9.5547002,7.16794971 L12.5547002,9.16794971 C13.1484333,9.56377175 13.1484333,10.4362282 12.5547002,10.8320503 L12.5547002,10.8320503 L9.5547002,12.8320503 C8.89014547,13.2750868 8,12.7986954 8,12 L8,12 Z" fill="#637381" fill-rule="nonzero" /></svg>
```
Comment (kaelig) [inline packages/polaris-icons-raw/icons/polaris/automation_major_monotone.svg]:
```suggestion
<svg viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path d="M12 0a1 1 0 011 1v1.582c.69.28 1.337.654 1.923 1.112l1.371-.792a1 1 0 011.366.366l2 3.464a1 1 0 01-.366 1.366l-1.37.792a8.062 8.062 0 010 2.22l1.37.792a1 1 0 01.366 1.366l-2 3.464a1 1 0 01-1.366.366l-1.371-.792A7.992 7.992 0 0113 17.418V19a1 1 0 01-1 1H8a1 1 0 01-1-1v-1.582a7.992 7.992 0 01-1.923-1.112l-1.371.792a1 1 0 01-1.366-.366l-2-3.464a1 1 0 01.366-1.366l1.37-.791a8.062 8.062 0 010-2.222L.707 8.1A1 1 0 01.34 6.731l2-3.464a1 1 0 011.366-.366l1.371.792a7.998 7.998 0 011.582-.965L7 2.582V1a1 1 0 011-1h4zm-1 2H9v1.29a1 1 0 01-.715.958A5.992 5.992 0 005.878 5.64a1 1 0 01-1.187.14l-1.12-.646-1 1.732 1.118.645a1 1 0 01.473 1.097 6.02 6.02 0 000 2.784 1 1 0 01-.473 1.097l-1.117.645 1 1.732 1.119-.646a1 1 0 011.187.14 5.992 5.992 0 002.407 1.392A1 1 0 019 16.71V18h2v-1.29a1 1 0 01.715-.959 5.992 5.992 0 002.407-1.391 1 1 0 011.188-.14l1.118.646 1-1.732-1.117-.645a1 1 0 01-.473-1.097 6.02 6.02 0 000-2.784 1 1 0 01.473-1.097l1.117-.645-1-1.732-1.118.646a1 1 0 01-1.188-.14 5.992 5.992 0 00-2.407-1.391 1 1 0 01-.714-.959V2zM8 8a1 1 0 011.555-.832l3 2a1 1 0 010 1.664l-3 2A1 1 0 018 12z" fill="#637381"/></svg>

```

File: packages/polaris-icons-raw/icons/polaris/automation_major_twotone.svg [hunk 5]
```diff
@@ -0,0 +1 @@
...3,15.877856 8.99994472,16.2676789 8.99995367,16.7100873 L8.99997977,18 L11.0002952,18 L11.0006762,16.7096248 C11.0008068,16.2673532 11.2914454,15.8776966 11.7153292,15.7514958 C12.6164265,15.4832164 13.4400914,15.0050211 14.122343,14.3596436 C14.4436729,14.0556809 14.92652,13.9989285 15.3095726,14.2201001 L16.428219,14.8659981 L17.4282045,13.1339724 L16.3109127,12.4889 C15.928172,12.2679233 15.7358769,11.8219096 15.8379535,11.3919079 C15.9452458,10.9399349 16,10.4739329 16,10 C16,9.52606708 15.9452458,9.06006506 15.8379535,8.60809214 C15.7358769,8.17809042 15.928172,7.73207674 16.3109127,7.51110004 L17.4282045,6.86602758 L16.428219,5.13400195 L15.3095726,5.77989986 C14.92652,6.00107153 14.4436729,5.94431911 14.122343,5.64035639 C13.4400914,4.99497892 12.6164265,4.51678356 11.7153292,4.24850424 C11.2914454,4.12230337 11.0008068,3.73264679 11.0006762,3.29037524 L11.0002952,2 Z M8,8 C8,7.20130462 8.89014547,6.72491322 9.5547002,7.16794971 L9.5547002,7.16794971 L12.5547002,9.16794971 C13.1484333,9.56377175 13.1484333,10.4362282 12.5547002,10.8320503 L12.5547002,10.8320503 L9.5547002,12.8320503 C8.89014547,13.2750868 8,12.7986954 8,12 L8,12 Z" fill="#637381" fill-rule="nonzero" /></svg>
```
Comment (kaelig) [inline packages/polaris-icons-raw/icons/polaris/automation_major_twotone.svg]:
```suggestion
<svg viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path d="M12 0a1 1 0 011 1v1.582c.69.28 1.337.654 1.923 1.112l1.371-.792a1 1 0 011.366.366l2 3.464a1 1 0 01-.366 1.366l-1.37.792a8.062 8.062 0 010 2.22l1.37.792a1 1 0 01.366 1.366l-2 3.464a1 1 0 01-1.366.366l-1.371-.792A7.992 7.992 0 0113 17.418V19a1 1 0 01-1 1H8a1 1 0 01-1-1v-1.582a7.992 7.992 0 01-1.923-1.112l-1.371.792a1 1 0 01-1.366-.366l-2-3.464a1 1 0 01.366-1.366l1.37-.791a8.062 8.062 0 010-2.222L.707 8.1A1 1 0 01.34 6.731l2-3.464a1 1 0 011.366-.366l1.371.792a7.998 7.998 0 011.582-.965L7 2.582V1a1 1 0 011-1h4zm-1 2H9v1.29a1 1 0 01-.715.958A5.992 5.992 0 005.878 5.64a1 1 0 01-1.187.14l-1.12-.646-1 1.732 1.118.645a1 1 0 01.473 1.097 6.02 6.02 0 000 2.784 1 1 0 01-.473 1.097l-1.117.645 1 1.732 1.119-.646a1 1 0 011.187.14 5.992 5.992 0 002.407 1.392A1 1 0 019 16.71V18h2v-1.29a1 1 0 01.715-.959 5.992 5.992 0 002.407-1.391 1 1 0 011.188-.14l1.118.646 1-1.732-1.117-.645a1 1 0 01-.473-1.097 6.02 6.02 0 000-2.784 1 1 0 01.473-1.097l1.117-.645-1-1.732-1.118.646a1 1 0 01-1.188-.14 5.992 5.992 0 00-2.407-1.391 1 1 0 01-.714-.959V2zM8 8a1 1 0 011.555-.832l3 2a1 1 0 010 1.664l-3 2A1 1 0 018 12z" fill="#637381"/></svg>

```

**Comment by kaelig** [comment]
I'm with you @amrocha, it's not good to have me or any human as a blocker for this kind of performance optimizations.

For this instance, I'll ask that we use optimized icons that I put as requested, and I have a plan to avoid this for the future: GitHub actions 🤖 

When we started this project, GitHub actions didn't exist, but now they do, so we can automate optimization at the PR level.

I've asked the author of the image optimization action we use on polaris-styleguide if they'd accept a PR to add SVGO support: https://github.com/calibreapp/image-actions/issues/7

I'll set up a meeting to discuss next steps when we have more clarity on whether this PR is welcome or if we should create our own action.

**Pull request merged**